python/ray/air/_internal/config.py

```diff
@@ -41,6 +19,15 @@ def ensure_only_allowed_dataclass_keys_updated(
 
     allowed_keys = set(allowed_keys)
 
+    keys_not_in_dict = [key for key in allowed_keys if key not in default_data.__dict__]
+    if keys_not_in_dict:
+        raise ValueError(
+            f"Key(s) {keys_not_in_dict} are not present in "
+            f"{dataclass.__class__.__name__}. "
+            "Remove them from `allowed_keys`. "
+            f"Valid keys: {list(default_data.__dict__.keys())}"
+        )
```

Comment:
Well, this will fail in the tests. There is no chance a user will see this error, unless the develop their own trainer (in which case that's actually good IMO. Makes it harder to shoot yourself in the foot)

Reply:
Yeah sounds reasonable. Can we update method name/docs? 

Longer term maybe we should split this logic out so that we have one validation step for the Trainer definition (developer), and one for the instantiation (end user).

Reply:
@matthewdeng done, ptal